From 90860d610e07732c01832c331b9a2b73578e71bb Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Wed, 16 Aug 2017 15:59:03 +0800
Subject: [PATCH 2381/5242] MLK-16197-6 video: fbdev: dcss: move
 'dcss_dtg_start' call to probe

commit  4adf7204b597b3d97644853fde7056e7b8fb5983 from
https://source.codeaurora.org/external/imx/linux-imx.git

Move the 'dcss_dtg_start' calling to probe stage which
can service the fifo commits generated in probe stage
in time.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/imx_dcss.c |   21 ++++++---------------
 1 file changed, 6 insertions(+), 15 deletions(-)

diff --git a/drivers/video/fbdev/mxc/imx_dcss.c b/drivers/video/fbdev/mxc/imx_dcss.c
index 0ab29b5..9c095d5 100644
--- a/drivers/video/fbdev/mxc/imx_dcss.c
+++ b/drivers/video/fbdev/mxc/imx_dcss.c
@@ -1959,6 +1959,8 @@ static int dcss_dtg_start(struct dcss_info *info)
 
 	writel(0xff000100, info->base + chans->dtg_addr + 0x0);
 
+	info->dcss_state = DCSS_STATE_RUNNING;
+
 	return 0;
 }
 
@@ -2813,21 +2815,6 @@ static int dcss_blank(int blank, struct fb_info *fbi)
 			goto out;
 		}
 #endif
-
-#if USE_CTXLD
-		if (!fb_node) {
-			/* start global timing */
-			if (info->dcss_state == DCSS_STATE_RESET) {
-				ret = dcss_dtg_start(info);
-				if (ret) {
-					dev_err(&pdev->dev, "start dtg failed\n");
-					goto out;
-				}
-
-				info->dcss_state = DCSS_STATE_RUNNING;
-			}
-		}
-#endif
 	} else {
 		dcss_channel_blank(blank, cinfo);
 #if USE_CTXLD
@@ -3253,6 +3240,10 @@ static int dcss_probe(struct platform_device *pdev)
 
 	dcss_interrupts_init(info);
 
+	ret = dcss_dtg_start(info);
+	if (ret)
+		goto kfree_info;
+
 	/* register channel 0: graphic */
 	ret = dcss_register_one_ch(0, info);
 	if (ret)
-- 
1.7.9.5

